packages/menu/src/MenuItem.ts

```diff
@@ -533,8 +530,7 @@ export class MenuItem extends LikeAnchor(Focusable) {
             return;
         }
         await new Promise((ready) => requestAnimationFrame(ready));
-        addOrUpdateEvent.reset(this);
-        this.dispatchEvent(addOrUpdateEvent);
+        this.dispatchEvent(new MenuItemAddedOrUpdatedEvent(this));
```

Comment:
We specifically do not want to `new` this event. Please find a path that does not require this.

Reply:
You can see the test case for this when this [changes the render time of Pickers under test by 4%](https://github.com/adobe/spectrum-web-components/pull/3189#user-content-picker).